### ISSUE TITLE
Output improvements & options

### ISSUE THREAD

**Issue by Ixohoxi** [mas-cli/mas#104]
This isn't a bug, but I don't see a way to offer suggestions other than using this interface. That said, I have a few suggestions about the formatting of output. It might be helpful to make the suggested additional columns display only via CLI switches.

- Display the product identifiers with right-justification—then the names of the programs would also line up nicely.
- Display the dates that each application was installed and last updated _on our system_.
- Display the dates that each application was added and last updated on the App Store, along with an indicator if the installed version can be updated.
- Display the required version of macOS.
- Sort the output by product identifier, product name, or one of the additional columns I've suggested.
- Retrieve the complete list of all applications—like how Homebrew behaves when "brew search" and "brew cask search" are run with no search string specified.

That's all the feedback I have for now. Many thanks for this amazing awesome tool!!!

**Comment by phatblat** [comment]
Adding more details to the columnar text output is tricky as wrapped lines become very difficult to read. However if we add alternative output formats (such as JSON #190) then the more details the marrier!

**Comment by rgoldberg** [comment]
@mas-cli/admins App IDs are right justified in some places, while left justified in others.

Can we choose one justification for app IDs (& other numbers), then consistently stick with it?

I prefer right justification.

**Comment by rgoldberg** [comment]
I'm closing this in favor of #1226, which is for a `--fields` option that allows field selection, field renaming, field ordering, field value formatting & row sorting. This will handle some (if not all) of the requested additional fields & sorting.

I've already right justified ADAM IDs (aka product identifiers).

I don't see a useful ROI for implementing any code to get any dates that aren't available from Spotlight for installed apps or from the iTunes Search web API for catalog apps (I haven't looked through if all the available dates cover the dates requested here). If anyone wants extra dates, they can submit a PR to provide them.

The App Store contains too many apps to listen to them all, and there is no easy way to get a list of all of them.

Thanks for the submission; most of the features here will be implemented after #1226 is released.
